File: Yet another query problem.java
class Solution {

    public static int freq(int n, int [] A, int i){

        int count = 0;

        for(int j=i; j<A.length; j++){

            if(A[j] == n){

                count++;

            }

        }

        return count;

    }

    public static ArrayList<Integer> solveQueries(int N, int num, int[] A, int[][] Q) {

        // code here

        ArrayList<Integer> list = new ArrayList<>();

        int row = num;

        int col = Q[0].length;

        for(int i=0; i<num; i++){

            int count=0;

            for(int j=Q[i][0]; j<=Q[i][1]; j++){

                if(freq(A[j], A, j) == Q[i][2]){

                    count++;

                }

            }

            list.add(count);

        }

        return list;

    }

}

File: Yet another query problem.cpp
class Solution {
  public:
    vector<int> solveQueries(int n, int num, vector<int> &a, vector<vector<int>> &Q) 
    {
       vector<int> v(n, 0);
       for(int i=0; i<n; i++)
       {
           int ele = a[i];
           int f = 0;
           for(int j=i; j<n; j++)
           {
               if(a[j]==ele)
               {
                   f++;
               }
           }
           v[i] = f;
       }
       
       vector<int> ans;
       for(int i=0; i<Q.size(); i++)
       {
           int l = Q[i][0];
           int r = Q[i][1];
           int k = Q[i][2];
           int a = 0;
           for(int j=l; j<=r; j++)
           {
               if(v[j]==k)
               {
                   a++;
               }
           }
           ans.push_back(a);
       }
       
       return ans;
    }
};
